### PR TITLE
fix: Set correct Content-Type when serving assets

### PR DESCRIPTION
**Pull request by sbernauer**
Users have reported problems using either Chrome, Edge or Safari getting the error message

> failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/octet-stream". Strict MIME type checking is enforced for module scripts per HTML spec.

The root cause was that we set the Content-Type header twice, once to application/octet-stream and the second to text/javascript. With this fix only a single (correct) Content-Type header is sent.

Thanks @Techassi for the help!

# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/stackable-cockpitd/src/handlers/ui.rs [hunk 1]
```diff
@@ -16,16 +16,20 @@ pub fn router() -> Router {
...
-            },
-        )]),
-        stackable_cockpit_web::ASSETS[&name],
-    )
+
+/// Adds (or replaces) the Content-Type header with the type of the served asset.
+/// So far only javascript and css are supported, for all the other types
+/// `application/octet-stream` will be used.
+async fn asset(mut headers: HeaderMap, Path(name): Path<String>) -> impl IntoResponse {
+    headers.insert(
```
Comment (nightkr) [inline rust/stackable-cockpitd/src/handlers/ui.rs:24]: This doesn't do what you want it to. The `HeaderMap` argument isn't some base set of headers defined by the framework, but the headers sent by the client. What this PR actually does is to reflect any headers sent by the user back at them.

File: rust/stackable-cockpitd/src/handlers/ui.rs [hunk 2]
```diff
@@ -16,16 +16,20 @@ pub fn router() -> Router {
...
-                Some((_, "css")) => HeaderValue::from_static("text/css"),
-                _ => HeaderValue::from_static("application/octet-stream"),
-            },
-        )]),
-        stackable_cockpit_web::ASSETS[&name],
-    )
+
+/// Adds (or replaces) the Content-Type header with the type of the served asset.
+/// So far only javascript and css are supported, for all the other types
+/// `application/octet-stream` will be used.
```
Comment (nightkr) [inline rust/stackable-cockpitd/src/handlers/ui.rs:22]: This is a detail of the function, but not its primary purpose. This should be either a subsection of the doccomment, or a regular comment inside of the function.

File: rust/stackable-cockpitd/src/handlers/ui.rs (hunk 1 above)
Comment (nightkr) [reply rust/stackable-cockpitd/src/handlers/ui.rs:24]: You can test this yourself: `curl 'http://localhost:8000/ui/assets/index-1272d2e9.js' -H 'foo: bar' -D- -o /dev/null | grep foo`.

**Review by nightkr — CHANGES_REQUESTED** [review]

**Comment by sbernauer** [comment]
Superseded by https://github.com/stackabletech/stackable-cockpit/pull/137

**Pull request closed without merging**